torch_xla/distributed/xla_multiprocessing.py

```diff
@@ -154,20 +154,28 @@ def spawn(fn,
       Default: `spawn`
 
   Returns:
-    The same object returned by the `torch.multiprocessing.spawn` API.
+    The same object returned by the `torch.multiprocessing.spawn` API. If
+    `nprocs` is 1 the `fn` function will be called directly, and the API will
+    return None.
   """
 
   if not _is_tpu_config():
     # If this is not an TPU setup, jump to normal multi-processing.
     nprocs = nprocs or 1
-    return torch.multiprocessing.spawn(
-        fn, args=args, nprocs=nprocs, join=join, daemon=daemon)
+    if nprocs == 1:
+      fn(0, *args)
```

Comment:
We should return here so that we don't run the below TPU part after running this, no? Or make the `_it_tpu_config()` an `if/else`?

Reply:
We better not return at all.